From f665fe3e2e74548a2a236f3b7635227621aa568a Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Wed, 9 Jan 2019 02:46:03 +0900
Subject: [PATCH 18/34] udev-node: make link_find_prioritized() return negative
 value when nothing found

Fixes a bug introduced by a2554acec652fc65c8ed0c6c1fede9ba8c3693b1.

Fixes RHBZ#1662303.

(cherry picked from commit 82d9ac23fd5ab2befe2a95187640a8d38799dd64)
---
 src/udev/udev-node.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/udev/udev-node.c b/src/udev/udev-node.c
index c11eb8c1ac..c77010c09b 100644
--- a/src/udev/udev-node.c
+++ b/src/udev/udev-node.c
@@ -181,6 +181,9 @@ static int link_find_prioritized(sd_device *dev, bool add, const char *stackdir,
                 priority = db_prio;
         }
 
+        if (!target)
+                return -ENOENT;
+
         *ret = TAKE_PTR(target);
         return 0;
 }
-- 
2.20.1

